### PR TITLE
CI: update github workflow config

### PR DESCRIPTION
**Pull request by 0xe3b0c4**
### Description

update github workflow config 

### Changes

Notable changes: 
* upgrade build environment to `ubuntu-22.04`
* upgrade golang to `1.17`
* add docker image build config

### CODE REVIEW

**Comment by 0xe3b0c4** [comment]
## remove old package require personal access token

1. [create personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) (need  `read:packages`  and  `delete:packages` ) 

scopes


2. Project setting add actions secret (secret name: `AUTO_DELETE_OLD_VERSION_PAT` )

setting

File: .github/workflows/build-devel-image.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+name: Build devel docker image
+
+on:
+  push:
+    branches:
+      - devel
+
+  pull_request:
+    types:
```
Comment (RichardBrown123go) [inline .github/workflows/build-devel-image.yml:11]: Add the branch is a better option.

**Review by RichardBrown123go — CHANGES_REQUESTED** [review]
LGTM
Only a few suggestion.

**Comment by RichardBrown123go** [comment]
@matterfun Would you follow this PR?
The CI needs to update some secrets in repo settings.

**Review by RichardBrown123go — APPROVED** [review]
Great job.
Thanks for your contribution.

**Pull request merged**